packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol

```diff
@@ -595,3 +595,81 @@ contract L1CrossDomainMessenger_Test is Bridge_Initializer {
         assertEq(l1CrossDomainMessenger.paused(), superchainConfig.paused());
     }
 }
+
+/// @dev A regression test against a reentrancy vulnerability in the CrossDomainMessenger contract, which
+///      was possible by intercepting and sandwhiching a signed Safe Transaction to upgrade it.
+contract L1CrossDomainMessenger_ReinitReentryTest is Bridge_Initializer {
+    bool attacked;
+
+    // Common values used across functions
+    address target;
+    address sender = sender = Predeploys.L2_CROSS_DOMAIN_MESSENGER;
+    uint256 messageValue = 50;
+
+    function setUp() public override {
+        super.setUp();
+        target = address(this);
+        vm.deal(address(l1CrossDomainMessenger), messageValue * 2);
+    }
+
+    /// @dev This method will be called by the relayed message, and will attempt to reenter the relayMessage function
+    ///      exactly once.
+    function reinitAndReenter() public payable {
+        // only attempt the attack once
+        if (!attacked) {
+            // set initialized to false
+            vm.store(address(l1CrossDomainMessenger), 0, bytes32(uint256(0)));
+
+            // call the initializer function
+            l1CrossDomainMessenger.initialize(SuperchainConfig(superchainConfig));
+
+            // attempt to re-replay the withdrawal
+            l1CrossDomainMessenger.relayMessage(
+                Encoding.encodeVersionedNonce({ _nonce: 0, _version: 1 }), // nonce
+                sender,
+                target,
+                messageValue,
+                0,
+                abi.encodeWithSelector(this.reinitAndReenter.selector)
+            );
+        }
+        attacked = true;
```

Comment:
I am confused that you dont need to set attacked=true before making the next call

Reply:
good catch. I had written the attack test previously and it was working, but I think I might have broken it at the same time as patching the bug. 

I'm addressing that by breaking out the fix in to a new pr #8863 so that it will have it's own CI run and will persist in git history.

Reply:
Resolved here: 
https://github.com/ethereum-optimism/optimism/pull/8863/files#r1443309343